When refreshing iptables, solve the interruption risk of communication services

yz271544 opened the pull request:
What type of PR is this?
/kind bug

What this PR does / why we need it:
Problems with accessing external services through edgemesh at the edge
Which issue(s) this PR fixes:

Fixes #71

Special notes for your reviewer:

    1. When getting edgemesh ignores routing rules and proxy rules, it also gets the routing rules that have been previously invalidated. These invalid routes need to be cleared first when refreshing the routing settings. Re-Ensure routing requires insert mode: `utiliptables.Prepend`
    2. add go-funk library
    3. add go-funk LICENSE
    4. upgrade the go-funk to v0.9.1
    5. modify proxyby!=edgemesh to noproxy=edgemesh


Signed-off-by: 胡正阳 huzhengyang@gridsum.com

yz271544 commented:
It seems that the go-funk library is not allowed to be added, would you consider adding it? Some collection operations can be provided

Poorunga commented:
need run: go mod vendor, then re-push your update.

yz271544 commented:
I executed these instructions, updated the vendor and LISENCE/vendor directories, and verified that I can succeed locally, and the error report on github made me wonder.
![push-0e19f14](https://user-images.githubusercontent.com/16499075/132320087-5932bd0d-cde1-401a-8676-7f77cd860be2.png)

Poorunga commented:
/cc @ZBoIsHere @khalid-jobs 
How to solve the problem of make verify-vendor failure, please take a look.

yz271544 commented:
[etl@etl09 edgemesh]$ go env
GO111MODULE="on"
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/etl/.cache/go-build"
GOENV="/home/etl/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE="gitlab.gridsum.com/iiotdepartment"
GONOPROXY="gitlab.gridsum.com/iiotdepartment"
GONOSUMDB="gitlab.gridsum.com/iiotdepartment"
GOOS="linux"
GOPATH="/lyndon/iProject/gopath"
GOPRIVATE="gitlab.gridsum.com/iiotdepartment"
GOPROXY="https://yz271544:UbFWoA20KLRShpM@goproxy.io,direct"
GOROOT="/home/etl/.g/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/etl/.g/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/lyndon/iProject/gopath/src/github.com/kubeedge/edgemesh/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build202794712=/tmp/go-build -gno-record-gcc-switches"
[etl@etl09 edgemesh]$ 
[etl@etl09 edgemesh]$ go version
go version go1.14.15 linux/amd64

yz271544 commented:
https://stackoverflow.com/questions/63521430/clear-cache-in-github-actions

Poorunga commented:
@yz271544 
I tested your branch on my own host and found that it can pass make verify.

You need to run these command:
1. go mod tidy
2. go mod download
3. go mod vendor
4. make verify (maybe need re-commit)
5. make lint (matbe need fix golang lint then re-commit)

Step4 and step5 may fail on your host, if it fails, please show me your log here.

Poorunga commented:
> https://stackoverflow.com/questions/63521430/clear-cache-in-github-actions

If you can run `make lint` and `make verify` on your own host and pass both of them, so can github actions. You need to ensure this first.

yz271544 commented:
thank you ! i try again, this is a logs:
```shell
[etl@etl09 edgemesh]$ go mod tidy
[etl@etl09 edgemesh]$ go mod download
[etl@etl09 edgemesh]$ go mod vendor
[etl@etl09 edgemesh]$ git status
On branch refresh-rules
nothing to commit, working tree clean

[etl@etl09 edgemesh]$ make verify
hack/verify-golang.sh
go detail version: go version go1.14.15 linux/amd64
go version: 1.14.15
hack/verify-vendor.sh
running 'go mod tidy'
running 'go mod vendor'
SUCCESS: Vendor Verified.
hack/verify-vendor-licenses.sh
bitbucket.org/bertimus9/systemstat doesn't exist in vendor, skipping
cloud.google.com/go doesn't exist in vendor, skipping
cloud.google.com/go/bigquery doesn't exist in vendor, skipping
cloud.google.com/go/datastore doesn't exist in vendor, skipping
cloud.google.com/go/pubsub doesn't exist in vendor, skipping
cloud.google.com/go/storage doesn't exist in vendor, skipping
dmitri.shuralyov.com/app/changes doesn't exist in vendor, skipping
dmitri.shuralyov.com/gpu/mtl doesn't exist in vendor, skipping
dmitri.shuralyov.com/html/belt doesn't exist in vendor, skipping
dmitri.shuralyov.com/service/change doesn't exist in vendor, skipping
dmitri.shuralyov.com/state doesn't exist in vendor, skipping
git.apache.org/thrift.git doesn't exist in vendor, skipping
github.com/256dpi/gomqtt doesn't exist in vendor, skipping
github.com/256dpi/mercury doesn't exist in vendor, skipping
github.com/abiosoft/ishell doesn't exist in vendor, skipping
github.com/abiosoft/readline doesn't exist in vendor, skipping
github.com/aead/siphash doesn't exist in vendor, skipping
github.com/agnivade/levenshtein doesn't exist in vendor, skipping
github.com/ajstarks/svgo doesn't exist in vendor, skipping
github.com/alecthomas/template doesn't exist in vendor, skipping
github.com/alecthomas/units doesn't exist in vendor, skipping
github.com/AndreasBriese/bbloom doesn't exist in vendor, skipping
github.com/andreyvit/diff doesn't exist in vendor, skipping
github.com/anmitsu/go-shlex doesn't exist in vendor, skipping
github.com/apache/servicecomb-kie doesn't exist in vendor, skipping
github.com/armon/circbuf doesn't exist in vendor, skipping
github.com/armon/consul-api doesn't exist in vendor, skipping
github.com/asaskevich/govalidator doesn't exist in vendor, skipping
github.com/astaxie/beego doesn't exist in vendor, skipping
github.com/auth0/go-jwt-middleware doesn't exist in vendor, skipping
github.com/aws/aws-sdk-go doesn't exist in vendor, skipping
github.com/Azure/azure-sdk-for-go doesn't exist in vendor, skipping
github.com/Azure/go-ansiterm
github.com/Azure/go-autorest/autorest doesn't exist in vendor, skipping
github.com/Azure/go-autorest/autorest/adal doesn't exist in vendor, skipping
github.com/Azure/go-autorest/autorest/date doesn't exist in vendor, skipping
github.com/Azure/go-autorest/autorest/mocks doesn't exist in vendor, skipping
github.com/Azure/go-autorest/autorest/to doesn't exist in vendor, skipping
github.com/Azure/go-autorest/autorest/validation doesn't exist in vendor, skipping
github.com/Azure/go-autorest/logger doesn't exist in vendor, skipping
github.com/Azure/go-autorest/tracing doesn't exist in vendor, skipping
github.com/beego/goyaml2 doesn't exist in vendor, skipping
github.com/beego/x2j doesn't exist in vendor, skipping
github.com/beorn7/perks doesn't exist in vendor, skipping
github.com/bgentry/speakeasy doesn't exist in vendor, skipping
github.com/bifurcation/mint doesn't exist in vendor, skipping
github.com/blang/semver doesn't exist in vendor, skipping
github.com/boltdb/bolt doesn't exist in vendor, skipping
github.com/bradfitz/go-smtpd doesn't exist in vendor, skipping
github.com/bradfitz/gomemcache doesn't exist in vendor, skipping
github.com/btcsuite/btcd
github.com/btcsuite/btclog doesn't exist in vendor, skipping
github.com/btcsuite/btcutil doesn't exist in vendor, skipping
github.com/btcsuite/go-socks doesn't exist in vendor, skipping
github.com/btcsuite/goleveldb doesn't exist in vendor, skipping
github.com/btcsuite/snappy-go doesn't exist in vendor, skipping
github.com/btcsuite/websocket doesn't exist in vendor, skipping
github.com/btcsuite/winsvc doesn't exist in vendor, skipping
github.com/buger/jsonparser doesn't exist in vendor, skipping
github.com/buraksezer/consistent
github.com/BurntSushi/toml doesn't exist in vendor, skipping
github.com/BurntSushi/xgb doesn't exist in vendor, skipping
github.com/caddyserver/caddy doesn't exist in vendor, skipping
github.com/casbin/casbin doesn't exist in vendor, skipping
github.com/cenkalti/backoff
github.com/census-instrumentation/opencensus-proto doesn't exist in vendor, skipping
github.com/cespare/xxhash, temporarily skipping
github.com/cespare/xxhash/v2 doesn't exist in vendor, skipping
github.com/chai2010/gettext-go doesn't exist in vendor, skipping
github.com/checkpoint-restore/go-criu/v4 doesn't exist in vendor, skipping
github.com/cheekybits/genny doesn't exist in vendor, skipping
github.com/chzyer/logex doesn't exist in vendor, skipping
github.com/chzyer/readline doesn't exist in vendor, skipping
github.com/chzyer/test doesn't exist in vendor, skipping
github.com/cilium/ebpf doesn't exist in vendor, skipping
github.com/client9/misspell doesn't exist in vendor, skipping
github.com/cloudflare/golz4 doesn't exist in vendor, skipping
github.com/clusterhq/flocker-go doesn't exist in vendor, skipping
github.com/cncf/udpa/go doesn't exist in vendor, skipping
github.com/cockroachdb/datadriven doesn't exist in vendor, skipping
github.com/codegangsta/negroni doesn't exist in vendor, skipping
github.com/container-storage-interface/spec doesn't exist in vendor, skipping
github.com/containerd/cgroups doesn't exist in vendor, skipping
github.com/containerd/console doesn't exist in vendor, skipping
github.com/containerd/containerd doesn't exist in vendor, skipping
github.com/containerd/continuity doesn't exist in vendor, skipping
github.com/containerd/fifo doesn't exist in vendor, skipping
github.com/containerd/go-runc doesn't exist in vendor, skipping
github.com/containerd/ttrpc doesn't exist in vendor, skipping
github.com/containerd/typeurl doesn't exist in vendor, skipping
github.com/containernetworking/cni doesn't exist in vendor, skipping
github.com/coredns/corefile-migration doesn't exist in vendor, skipping
github.com/coreos/bbolt doesn't exist in vendor, skipping
github.com/coreos/etcd doesn't exist in vendor, skipping
github.com/coreos/go-etcd doesn't exist in vendor, skipping
github.com/coreos/go-oidc doesn't exist in vendor, skipping
github.com/coreos/go-semver doesn't exist in vendor, skipping
github.com/coreos/go-systemd doesn't exist in vendor, skipping
github.com/coreos/go-systemd/v22 doesn't exist in vendor, skipping
github.com/coreos/pkg doesn't exist in vendor, skipping
github.com/couchbase/go-couchbase doesn't exist in vendor, skipping
github.com/couchbase/gomemcached doesn't exist in vendor, skipping
github.com/couchbase/goutils doesn't exist in vendor, skipping
github.com/cpuguy83/go-md2man doesn't exist in vendor, skipping
github.com/cpuguy83/go-md2man/v2 doesn't exist in vendor, skipping
github.com/creack/pty doesn't exist in vendor, skipping
github.com/cupcake/rdb doesn't exist in vendor, skipping
github.com/cyphar/filepath-securejoin doesn't exist in vendor, skipping
github.com/davecgh/go-spew
github.com/daviddengcn/go-colortext doesn't exist in vendor, skipping
github.com/davidlazar/go-crypto
github.com/dgraph-io/badger doesn't exist in vendor, skipping
github.com/dgraph-io/ristretto doesn't exist in vendor, skipping
github.com/dgrijalva/jwt-go
github.com/dgryski/go-farm doesn't exist in vendor, skipping
github.com/dgryski/go-sip13 doesn't exist in vendor, skipping
github.com/dnaeon/go-vcr doesn't exist in vendor, skipping
github.com/docker/distribution doesn't exist in vendor, skipping
github.com/docker/docker doesn't exist in vendor, skipping
github.com/docker/go-connections doesn't exist in vendor, skipping
github.com/docker/go-units doesn't exist in vendor, skipping
github.com/docker/spdystream doesn't exist in vendor, skipping
github.com/docopt/docopt-go doesn't exist in vendor, skipping
github.com/dustin/go-humanize doesn't exist in vendor, skipping
github.com/eclipse/paho.mqtt.golang
github.com/edsrzf/mmap-go doesn't exist in vendor, skipping
github.com/elazarl/go-bindata-assetfs doesn't exist in vendor, skipping
github.com/elazarl/goproxy doesn't exist in vendor, skipping
github.com/emicklei/go-restful
github.com/envoyproxy/go-control-plane doesn't exist in vendor, skipping
github.com/envoyproxy/protoc-gen-validate doesn't exist in vendor, skipping
github.com/euank/go-kmsg-parser doesn't exist in vendor, skipping
github.com/evanphx/json-patch doesn't exist in vendor, skipping
github.com/exponent-io/jsonpath doesn't exist in vendor, skipping
github.com/fatih/camelcase doesn't exist in vendor, skipping
github.com/fatih/color doesn't exist in vendor, skipping
github.com/flynn-archive/go-shlex doesn't exist in vendor, skipping
github.com/flynn/go-shlex doesn't exist in vendor, skipping
github.com/flynn/noise
github.com/fogleman/gg doesn't exist in vendor, skipping
github.com/francoispqt/gojay doesn't exist in vendor, skipping
github.com/fsnotify/fsnotify
github.com/ghodss/yaml
github.com/gliderlabs/ssh doesn't exist in vendor, skipping
github.com/globalsign/mgo doesn't exist in vendor, skipping
github.com/go-acme/lego doesn't exist in vendor, skipping
github.com/go-bindata/go-bindata doesn't exist in vendor, skipping
github.com/go-chassis/foundation
github.com/go-chassis/go-archaius
github.com/go-chassis/go-chassis
github.com/go-chassis/go-chassis-config
github.com/go-chassis/go-restful-swagger20
github.com/go-chassis/paas-lager doesn't exist in vendor, skipping
github.com/go-check/check doesn't exist in vendor, skipping
github.com/go-errors/errors doesn't exist in vendor, skipping
github.com/go-gl/glfw/v3.3/glfw doesn't exist in vendor, skipping
github.com/go-ini/ini doesn't exist in vendor, skipping
github.com/go-kit/kit doesn't exist in vendor, skipping
github.com/go-logfmt/logfmt doesn't exist in vendor, skipping
github.com/go-logr/logr
github.com/go-mesh/openlogging
github.com/go-ole/go-ole doesn't exist in vendor, skipping
github.com/go-openapi/analysis doesn't exist in vendor, skipping
github.com/go-openapi/errors doesn't exist in vendor, skipping
github.com/go-openapi/jsonpointer doesn't exist in vendor, skipping
github.com/go-openapi/jsonreference doesn't exist in vendor, skipping
github.com/go-openapi/loads doesn't exist in vendor, skipping
github.com/go-openapi/runtime doesn't exist in vendor, skipping
github.com/go-openapi/spec doesn't exist in vendor, skipping
github.com/go-openapi/strfmt doesn't exist in vendor, skipping
github.com/go-openapi/swag doesn't exist in vendor, skipping
github.com/go-openapi/validate doesn't exist in vendor, skipping
github.com/go-ozzo/ozzo-validation doesn't exist in vendor, skipping
github.com/go-redis/redis doesn't exist in vendor, skipping
github.com/go-sql-driver/mysql doesn't exist in vendor, skipping
github.com/go-stack/stack doesn't exist in vendor, skipping
github.com/go-task/slim-sprig doesn't exist in vendor, skipping
github.com/goburrow/serial doesn't exist in vendor, skipping
github.com/godbus/dbus/v5 doesn't exist in vendor, skipping
github.com/gogo/protobuf
github.com/golang/freetype doesn't exist in vendor, skipping
github.com/golang/glog doesn't exist in vendor, skipping
github.com/golang/groupcache
github.com/golang/lint doesn't exist in vendor, skipping
github.com/golang/mock doesn't exist in vendor, skipping
github.com/golang/protobuf
github.com/golang/snappy doesn't exist in vendor, skipping
github.com/golangplus/bytes doesn't exist in vendor, skipping
github.com/golangplus/fmt doesn't exist in vendor, skipping
github.com/golangplus/testing doesn't exist in vendor, skipping
github.com/gomodule/redigo doesn't exist in vendor, skipping
github.com/google/btree doesn't exist in vendor, skipping
github.com/google/cadvisor doesn't exist in vendor, skipping
github.com/google/go-cmp
github.com/google/go-github doesn't exist in vendor, skipping
github.com/google/go-querystring doesn't exist in vendor, skipping
github.com/google/gofuzz
github.com/google/gopacket
github.com/google/martian doesn't exist in vendor, skipping
github.com/google/pprof doesn't exist in vendor, skipping
github.com/google/renameio doesn't exist in vendor, skipping
github.com/google/uuid
github.com/googleapis/gax-go doesn't exist in vendor, skipping
github.com/googleapis/gax-go/v2 doesn't exist in vendor, skipping
github.com/googleapis/gnostic
github.com/GoogleCloudPlatform/k8s-cloud-provider doesn't exist in vendor, skipping
github.com/gophercloud/gophercloud doesn't exist in vendor, skipping
github.com/gopherjs/gopherjs doesn't exist in vendor, skipping
github.com/gorilla/context doesn't exist in vendor, skipping
github.com/gorilla/mux
github.com/gorilla/websocket
github.com/gregjones/httpcache doesn't exist in vendor, skipping
github.com/grpc-ecosystem/go-grpc-middleware doesn't exist in vendor, skipping
github.com/grpc-ecosystem/go-grpc-prometheus doesn't exist in vendor, skipping
github.com/grpc-ecosystem/grpc-gateway doesn't exist in vendor, skipping
github.com/gxed/hashland/keccakpg doesn't exist in vendor, skipping
github.com/gxed/hashland/murmur3 doesn't exist in vendor, skipping
github.com/hashicorp/go-syslog doesn't exist in vendor, skipping
github.com/hashicorp/go-version
github.com/hashicorp/golang-lru
github.com/hashicorp/hcl doesn't exist in vendor, skipping
github.com/heketi/heketi doesn't exist in vendor, skipping
github.com/heketi/tests doesn't exist in vendor, skipping
github.com/hpcloud/tail doesn't exist in vendor, skipping
github.com/huin/goupnp
github.com/huin/goutil doesn't exist in vendor, skipping
github.com/ianlancetaylor/demangle doesn't exist in vendor, skipping
github.com/imdario/mergo
github.com/inconshreveable/mousetrap
github.com/ipfs/go-cid
github.com/ipfs/go-datastore doesn't exist in vendor, skipping
github.com/ipfs/go-detect-race doesn't exist in vendor, skipping
github.com/ipfs/go-ds-badger doesn't exist in vendor, skipping
github.com/ipfs/go-ds-leveldb doesn't exist in vendor, skipping
github.com/ipfs/go-ipfs-delay doesn't exist in vendor, skipping
github.com/ipfs/go-ipfs-util
github.com/ipfs/go-log
github.com/ipfs/go-log/v2
github.com/ishidawataru/sctp doesn't exist in vendor, skipping
github.com/jackpal/gateway doesn't exist in vendor, skipping
github.com/jackpal/go-nat-pmp
github.com/jbenet/go-cienv doesn't exist in vendor, skipping
github.com/jbenet/go-temp-err-catcher
github.com/jbenet/goprocess
github.com/JeffAshton/win_pdh doesn't exist in vendor, skipping
github.com/jellevandenhooff/dkim doesn't exist in vendor, skipping
github.com/jessevdk/go-flags doesn't exist in vendor, skipping
github.com/jimstudt/http-authentication doesn't exist in vendor, skipping
github.com/jmespath/go-jmespath doesn't exist in vendor, skipping
github.com/jonboulle/clockwork doesn't exist in vendor, skipping
github.com/jpillora/backoff doesn't exist in vendor, skipping
github.com/jrick/logrotate doesn't exist in vendor, skipping
github.com/json-iterator/go
github.com/jstemmer/go-junit-report doesn't exist in vendor, skipping
github.com/jtolds/gls doesn't exist in vendor, skipping
github.com/juju/ratelimit doesn't exist in vendor, skipping
github.com/julienschmidt/httprouter doesn't exist in vendor, skipping
github.com/jung-kurt/gofpdf doesn't exist in vendor, skipping
github.com/kami-zh/go-capturer doesn't exist in vendor, skipping
github.com/karrick/godirwalk doesn't exist in vendor, skipping
github.com/kisielk/errcheck doesn't exist in vendor, skipping
github.com/kisielk/gotool doesn't exist in vendor, skipping
github.com/kkdai/bstream doesn't exist in vendor, skipping
github.com/klauspost/cpuid doesn't exist in vendor, skipping
github.com/Knetic/govaluate doesn't exist in vendor, skipping
github.com/konsorten/go-windows-terminal-sequences
github.com/koron/go-ssdp
github.com/kr/logfmt doesn't exist in vendor, skipping
github.com/kr/pretty doesn't exist in vendor, skipping
github.com/kr/pty doesn't exist in vendor, skipping
github.com/kr/text doesn't exist in vendor, skipping
github.com/kubeedge/beehive
github.com/kubeedge/edgemesh doesn't exist in vendor, skipping
github.com/kubeedge/kubeedge
github.com/kubeedge/viaduct
github.com/kubernetes-csi/csi-lib-utils doesn't exist in vendor, skipping
github.com/Kubuxu/go-os-helper doesn't exist in vendor, skipping
github.com/kylelemons/godebug doesn't exist in vendor, skipping
github.com/lib/pq doesn't exist in vendor, skipping
github.com/libopenstorage/openstorage doesn't exist in vendor, skipping
github.com/libp2p/go-addr-util
github.com/libp2p/go-buffer-pool
github.com/libp2p/go-conn-security-multistream
github.com/libp2p/go-eventbus
github.com/libp2p/go-flow-metrics
github.com/libp2p/go-libp2p
github.com/libp2p/go-libp2p-autonat
github.com/libp2p/go-libp2p-blankhost
github.com/libp2p/go-libp2p-circuit
github.com/libp2p/go-libp2p-core
github.com/libp2p/go-libp2p-crypto doesn't exist in vendor, skipping
github.com/libp2p/go-libp2p-discovery
github.com/libp2p/go-libp2p-loggables
github.com/libp2p/go-libp2p-mplex
github.com/libp2p/go-libp2p-nat
github.com/libp2p/go-libp2p-netutil doesn't exist in vendor, skipping
github.com/libp2p/go-libp2p-noise, temporarily skipping
github.com/libp2p/go-libp2p-peer doesn't exist in vendor, skipping
github.com/libp2p/go-libp2p-peerstore
github.com/libp2p/go-libp2p-pnet
github.com/libp2p/go-libp2p-quic-transport doesn't exist in vendor, skipping
github.com/libp2p/go-libp2p-secio doesn't exist in vendor, skipping
github.com/libp2p/go-libp2p-swarm
github.com/libp2p/go-libp2p-testing doesn't exist in vendor, skipping
github.com/libp2p/go-libp2p-tls
github.com/libp2p/go-libp2p-transport-upgrader
github.com/libp2p/go-libp2p-yamux
github.com/libp2p/go-maddr-filter doesn't exist in vendor, skipping
github.com/libp2p/go-mplex
github.com/libp2p/go-msgio
github.com/libp2p/go-nat
github.com/libp2p/go-netroute
github.com/libp2p/go-openssl
github.com/libp2p/go-reuseport
github.com/libp2p/go-reuseport-transport
github.com/libp2p/go-sockaddr
github.com/libp2p/go-stream-muxer doesn't exist in vendor, skipping
github.com/libp2p/go-stream-muxer-multistream
github.com/libp2p/go-tcp-transport
github.com/libp2p/go-ws-transport
github.com/libp2p/go-yamux, temporarily skipping
github.com/libp2p/go-yamux/v2
github.com/liggitt/tabwriter doesn't exist in vendor, skipping
github.com/lithammer/dedent doesn't exist in vendor, skipping
github.com/lpabon/godbc doesn't exist in vendor, skipping
github.com/lucas-clemente/aes12 doesn't exist in vendor, skipping
github.com/lucas-clemente/quic-clients doesn't exist in vendor, skipping
github.com/lucas-clemente/quic-go doesn't exist in vendor, skipping
github.com/lucas-clemente/quic-go-certificates doesn't exist in vendor, skipping
github.com/lunixbochs/vtclean doesn't exist in vendor, skipping
github.com/magiconair/properties doesn't exist in vendor, skipping
github.com/mailru/easyjson doesn't exist in vendor, skipping
github.com/MakeNowJust/heredoc doesn't exist in vendor, skipping
github.com/marten-seemann/qpack doesn't exist in vendor, skipping
github.com/marten-seemann/qtls doesn't exist in vendor, skipping
github.com/marten-seemann/qtls-go1-15 doesn't exist in vendor, skipping
github.com/mattn/go-colorable doesn't exist in vendor, skipping
github.com/mattn/go-isatty doesn't exist in vendor, skipping
github.com/mattn/go-runewidth doesn't exist in vendor, skipping
github.com/mattn/go-sqlite3 doesn't exist in vendor, skipping
github.com/matttproud/golang_protobuf_extensions doesn't exist in vendor, skipping
github.com/mgutz/ansi doesn't exist in vendor, skipping
github.com/mholt/certmagic doesn't exist in vendor, skipping
github.com/microcosm-cc/bluemonday doesn't exist in vendor, skipping
github.com/Microsoft/go-winio doesn't exist in vendor, skipping
github.com/Microsoft/hcsshim doesn't exist in vendor, skipping
github.com/miekg/dns
github.com/mindprince/gonvml doesn't exist in vendor, skipping
github.com/minio/blake2b-simd
github.com/minio/sha256-simd
github.com/mistifyio/go-zfs doesn't exist in vendor, skipping
github.com/mitchellh/go-homedir doesn't exist in vendor, skipping
github.com/mitchellh/go-ps doesn't exist in vendor, skipping
github.com/mitchellh/go-wordwrap doesn't exist in vendor, skipping
github.com/mitchellh/mapstructure doesn't exist in vendor, skipping
github.com/moby/ipvs doesn't exist in vendor, skipping
github.com/moby/sys/mountinfo doesn't exist in vendor, skipping
github.com/moby/term
github.com/modern-go/concurrent
github.com/modern-go/reflect2
github.com/mohae/deepcopy doesn't exist in vendor, skipping
github.com/morikuni/aec doesn't exist in vendor, skipping
github.com/mr-tron/base58
github.com/mrunalp/fileutils doesn't exist in vendor, skipping
github.com/multiformats/go-base32
github.com/multiformats/go-base36
github.com/multiformats/go-multiaddr
github.com/multiformats/go-multiaddr-dns
github.com/multiformats/go-multiaddr-fmt
github.com/multiformats/go-multiaddr-net
github.com/multiformats/go-multibase
github.com/multiformats/go-multihash
github.com/multiformats/go-multistream
github.com/multiformats/go-varint
github.com/munnerz/goautoneg doesn't exist in vendor, skipping
github.com/mvdan/xurls doesn't exist in vendor, skipping
github.com/mwitkow/go-conntrack doesn't exist in vendor, skipping
github.com/mxk/go-flowrate doesn't exist in vendor, skipping
github.com/naoina/go-stringutil doesn't exist in vendor, skipping
github.com/naoina/toml doesn't exist in vendor, skipping
github.com/neelance/astrewrite doesn't exist in vendor, skipping
github.com/neelance/sourcemap doesn't exist in vendor, skipping
github.com/nxadm/tail
github.com/NYTimes/gziphandler doesn't exist in vendor, skipping
github.com/oklog/ulid doesn't exist in vendor, skipping
github.com/olekukonko/tablewriter doesn't exist in vendor, skipping
github.com/OneOfOne/xxhash doesn't exist in vendor, skipping
github.com/onsi/ginkgo
github.com/onsi/gomega
github.com/opencontainers/go-digest doesn't exist in vendor, skipping
github.com/opencontainers/image-spec doesn't exist in vendor, skipping
github.com/opencontainers/runc doesn't exist in vendor, skipping
github.com/opencontainers/runtime-spec doesn't exist in vendor, skipping
github.com/opencontainers/selinux doesn't exist in vendor, skipping
github.com/opentracing/opentracing-go
github.com/openzipkin/zipkin-go doesn't exist in vendor, skipping
github.com/OwnLocal/goes doesn't exist in vendor, skipping
github.com/patrickmn/go-cache
github.com/paypal/gatt doesn't exist in vendor, skipping
github.com/pborman/uuid doesn't exist in vendor, skipping
github.com/pelletier/go-toml doesn't exist in vendor, skipping
github.com/peterbourgon/diskv doesn't exist in vendor, skipping
github.com/pkg/errors
github.com/pmezard/go-difflib doesn't exist in vendor, skipping
github.com/pquerna/cachecontrol doesn't exist in vendor, skipping
github.com/prometheus/client_golang doesn't exist in vendor, skipping
github.com/prometheus/client_model doesn't exist in vendor, skipping
github.com/prometheus/common doesn't exist in vendor, skipping
github.com/prometheus/procfs doesn't exist in vendor, skipping
github.com/prometheus/tsdb doesn't exist in vendor, skipping
github.com/PuerkitoBio/purell doesn't exist in vendor, skipping
github.com/PuerkitoBio/urlesc doesn't exist in vendor, skipping
github.com/quobyte/api doesn't exist in vendor, skipping
github.com/remyoudompheng/bigfft doesn't exist in vendor, skipping
github.com/robfig/cron doesn't exist in vendor, skipping
github.com/rogpeppe/fastuuid doesn't exist in vendor, skipping
github.com/rogpeppe/go-internal doesn't exist in vendor, skipping
github.com/rubiojr/go-vhd doesn't exist in vendor, skipping
github.com/russross/blackfriday doesn't exist in vendor, skipping
github.com/russross/blackfriday/v2 doesn't exist in vendor, skipping
github.com/sailorvii/modbus doesn't exist in vendor, skipping
github.com/satori/go.uuid
github.com/seccomp/libseccomp-golang doesn't exist in vendor, skipping
github.com/sergi/go-diff doesn't exist in vendor, skipping
github.com/shirou/gopsutil doesn't exist in vendor, skipping
github.com/shurcooL/component doesn't exist in vendor, skipping
github.com/shurcooL/events doesn't exist in vendor, skipping
github.com/shurcooL/github_flavored_markdown doesn't exist in vendor, skipping
github.com/shurcooL/go doesn't exist in vendor, skipping
github.com/shurcooL/go-goon doesn't exist in vendor, skipping
github.com/shurcooL/gofontwoff doesn't exist in vendor, skipping
github.com/shurcooL/gopherjslib doesn't exist in vendor, skipping
github.com/shurcooL/highlight_diff doesn't exist in vendor, skipping
github.com/shurcooL/highlight_go doesn't exist in vendor, skipping
github.com/shurcooL/home doesn't exist in vendor, skipping
github.com/shurcooL/htmlg doesn't exist in vendor, skipping
github.com/shurcooL/httperror doesn't exist in vendor, skipping
github.com/shurcooL/httpfs doesn't exist in vendor, skipping
github.com/shurcooL/httpgzip doesn't exist in vendor, skipping
github.com/shurcooL/issues doesn't exist in vendor, skipping
github.com/shurcooL/issuesapp doesn't exist in vendor, skipping
github.com/shurcooL/notifications doesn't exist in vendor, skipping
github.com/shurcooL/octicon doesn't exist in vendor, skipping
github.com/shurcooL/reactions doesn't exist in vendor, skipping
github.com/shurcooL/sanitized_anchor_name doesn't exist in vendor, skipping
github.com/shurcooL/users doesn't exist in vendor, skipping
github.com/shurcooL/webdavfs doesn't exist in vendor, skipping
github.com/siddontang/go doesn't exist in vendor, skipping
github.com/siddontang/ledisdb doesn't exist in vendor, skipping
github.com/siddontang/rdb doesn't exist in vendor, skipping
github.com/sirupsen/logrus
github.com/smartystreets/assertions doesn't exist in vendor, skipping
github.com/smartystreets/goconvey doesn't exist in vendor, skipping
github.com/smola/gocompat doesn't exist in vendor, skipping
github.com/soheilhy/cmux doesn't exist in vendor, skipping
github.com/sourcegraph/annotate doesn't exist in vendor, skipping
github.com/sourcegraph/syntaxhighlight doesn't exist in vendor, skipping
github.com/spacemonkeygo/openssl doesn't exist in vendor, skipping
github.com/spacemonkeygo/spacelog
github.com/spaolacci/murmur3
github.com/spf13/afero doesn't exist in vendor, skipping
github.com/spf13/cast
github.com/spf13/cobra
github.com/spf13/jwalterweatherman doesn't exist in vendor, skipping
github.com/spf13/pflag
github.com/spf13/viper doesn't exist in vendor, skipping
github.com/src-d/envconfig doesn't exist in vendor, skipping
github.com/ssdb/gossdb doesn't exist in vendor, skipping
github.com/StackExchange/wmi doesn't exist in vendor, skipping
github.com/storageos/go-api doesn't exist in vendor, skipping
github.com/stretchr/objx doesn't exist in vendor, skipping
github.com/stretchr/testify doesn't exist in vendor, skipping
github.com/syndtr/gocapability doesn't exist in vendor, skipping
github.com/syndtr/goleveldb doesn't exist in vendor, skipping
github.com/tarm/serial doesn't exist in vendor, skipping
github.com/thecodeteam/goscaleio doesn't exist in vendor, skipping
github.com/thoas/go-funk
github.com/tidwall/pretty doesn't exist in vendor, skipping
github.com/tmc/grpc-websocket-proxy doesn't exist in vendor, skipping
github.com/ugorji/go doesn't exist in vendor, skipping
github.com/ugorji/go/codec doesn't exist in vendor, skipping
github.com/urfave/cli doesn't exist in vendor, skipping
github.com/urfave/negroni doesn't exist in vendor, skipping
github.com/vektah/gqlparser doesn't exist in vendor, skipping
github.com/viant/assertly doesn't exist in vendor, skipping
github.com/viant/toolbox doesn't exist in vendor, skipping
github.com/vishvananda/netlink
github.com/vishvananda/netns
github.com/vmware/govmomi doesn't exist in vendor, skipping
github.com/wendal/errors doesn't exist in vendor, skipping
github.com/whyrusleeping/go-keyspace doesn't exist in vendor, skipping
github.com/whyrusleeping/go-logging doesn't exist in vendor, skipping
github.com/whyrusleeping/mafmt doesn't exist in vendor, skipping
github.com/whyrusleeping/mdns doesn't exist in vendor, skipping
github.com/whyrusleeping/multiaddr-filter
github.com/x-cray/logrus-prefixed-formatter doesn't exist in vendor, skipping
github.com/xdg/scram doesn't exist in vendor, skipping
github.com/xdg/stringprep doesn't exist in vendor, skipping
github.com/xiang90/probing doesn't exist in vendor, skipping
github.com/xlab/handysort doesn't exist in vendor, skipping
github.com/xordataexchange/crypt doesn't exist in vendor, skipping
github.com/yuin/goldmark doesn't exist in vendor, skipping
go.etcd.io/bbolt doesn't exist in vendor, skipping
go.etcd.io/etcd doesn't exist in vendor, skipping
go.mongodb.org/mongo-driver doesn't exist in vendor, skipping
go.opencensus.io
go.uber.org/atomic
go.uber.org/goleak doesn't exist in vendor, skipping
go.uber.org/multierr
go.uber.org/ratelimit
go.uber.org/tools doesn't exist in vendor, skipping
go.uber.org/zap
go4.org doesn't exist in vendor, skipping
golang.org/x/build doesn't exist in vendor, skipping
golang.org/x/crypto
golang.org/x/exp doesn't exist in vendor, skipping
golang.org/x/image doesn't exist in vendor, skipping
golang.org/x/lint doesn't exist in vendor, skipping
golang.org/x/mobile doesn't exist in vendor, skipping
golang.org/x/mod doesn't exist in vendor, skipping
golang.org/x/net
golang.org/x/oauth2
golang.org/x/perf doesn't exist in vendor, skipping
golang.org/x/sync doesn't exist in vendor, skipping
golang.org/x/sys
golang.org/x/term doesn't exist in vendor, skipping
golang.org/x/text
golang.org/x/time
golang.org/x/tools doesn't exist in vendor, skipping
golang.org/x/xerrors
gonum.org/v1/gonum doesn't exist in vendor, skipping
gonum.org/v1/netlib doesn't exist in vendor, skipping
gonum.org/v1/plot doesn't exist in vendor, skipping
google.golang.org/api doesn't exist in vendor, skipping
google.golang.org/appengine
google.golang.org/genproto doesn't exist in vendor, skipping
google.golang.org/grpc doesn't exist in vendor, skipping
google.golang.org/protobuf
gopkg.in/alecthomas/kingpin.v2 doesn't exist in vendor, skipping
gopkg.in/check.v1 doesn't exist in vendor, skipping
gopkg.in/cheggaaa/pb.v1 doesn't exist in vendor, skipping
gopkg.in/errgo.v2 doesn't exist in vendor, skipping
gopkg.in/fsnotify.v1 doesn't exist in vendor, skipping
gopkg.in/gcfg.v1 doesn't exist in vendor, skipping
gopkg.in/go-playground/assert.v1 doesn't exist in vendor, skipping
gopkg.in/inf.v0
gopkg.in/mcuadros/go-syslog.v2 doesn't exist in vendor, skipping
gopkg.in/natefinch/lumberjack.v2 doesn't exist in vendor, skipping
gopkg.in/resty.v1 doesn't exist in vendor, skipping
gopkg.in/square/go-jose.v2 doesn't exist in vendor, skipping
gopkg.in/src-d/go-cli.v0 doesn't exist in vendor, skipping
gopkg.in/src-d/go-log.v1 doesn't exist in vendor, skipping
gopkg.in/tomb.v1
gopkg.in/tomb.v2 doesn't exist in vendor, skipping
gopkg.in/warnings.v0 doesn't exist in vendor, skipping
gopkg.in/yaml.v2
gopkg.in/yaml.v3 doesn't exist in vendor, skipping
gotest.tools doesn't exist in vendor, skipping
gotest.tools/v3 doesn't exist in vendor, skipping
grpc.go4.org doesn't exist in vendor, skipping
honnef.co/go/tools doesn't exist in vendor, skipping
istio.io/api
istio.io/client-go
istio.io/gogo-genproto
k8s.io/api
k8s.io/apiextensions-apiserver doesn't exist in vendor, skipping
k8s.io/apimachinery
k8s.io/apiserver
k8s.io/cli-runtime doesn't exist in vendor, skipping
k8s.io/client-go
k8s.io/cloud-provider doesn't exist in vendor, skipping
k8s.io/cluster-bootstrap doesn't exist in vendor, skipping
k8s.io/code-generator doesn't exist in vendor, skipping
k8s.io/component-base
k8s.io/cri-api doesn't exist in vendor, skipping
k8s.io/csi-translation-lib doesn't exist in vendor, skipping
k8s.io/gengo doesn't exist in vendor, skipping
k8s.io/heapster doesn't exist in vendor, skipping
k8s.io/klog, temporarily skipping
k8s.io/klog/v2
k8s.io/kube-aggregator doesn't exist in vendor, skipping
k8s.io/kube-controller-manager doesn't exist in vendor, skipping
k8s.io/kube-openapi
k8s.io/kube-proxy doesn't exist in vendor, skipping
k8s.io/kube-scheduler doesn't exist in vendor, skipping
k8s.io/kubectl doesn't exist in vendor, skipping
k8s.io/kubelet doesn't exist in vendor, skipping
k8s.io/kubernetes
k8s.io/legacy-cloud-providers doesn't exist in vendor, skipping
k8s.io/metrics doesn't exist in vendor, skipping
k8s.io/sample-apiserver doesn't exist in vendor, skipping
k8s.io/system-validators doesn't exist in vendor, skipping
k8s.io/utils
modernc.org/cc doesn't exist in vendor, skipping
modernc.org/golex doesn't exist in vendor, skipping
modernc.org/mathutil doesn't exist in vendor, skipping
modernc.org/strutil doesn't exist in vendor, skipping
modernc.org/xc doesn't exist in vendor, skipping
rsc.io/binaryregexp doesn't exist in vendor, skipping
rsc.io/pdf doesn't exist in vendor, skipping
rsc.io/quote/v3 doesn't exist in vendor, skipping
rsc.io/sampler doesn't exist in vendor, skipping
sigs.k8s.io/apiserver-network-proxy/konnectivity-client doesn't exist in vendor, skipping
sigs.k8s.io/kustomize doesn't exist in vendor, skipping
sigs.k8s.io/structured-merge-diff/v3 doesn't exist in vendor, skipping
sigs.k8s.io/structured-merge-diff/v4
sigs.k8s.io/yaml
sourcegraph.com/sourcegraph/go-diff doesn't exist in vendor, skipping
sourcegraph.com/sqs/pbtypes doesn't exist in vendor, skipping
vbom.ml/util doesn't exist in vendor, skipping

[etl@etl09 edgemesh]$ make lint
hack/make-rules/lint.sh
checking golangci-lint
found golangci-lint, version: golangci-lint has version 1.42.1 built from 54f4301d on 2021-09-06T17:05:10Z
start lint ...
check any whitenoise ...
check any issue by golangci-lint ...
INFO [config_reader] Config search paths: [./ /lyndon/iProject/gopath/src/github.com/kubeedge/edgemesh /lyndon/iProject/gopath/src/github.com/kubeedge /lyndon/iProject/gopath/src/github.com /lyndon/iProject/gopath/src /lyndon/iProject/gopath /lyndon/iProject /lyndon / /home/etl] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 8 linters: [errcheck gci goconst gofmt gosimple govet misspell whitespace] 
INFO [loader] Go packages loading at mode 575 (deps|files|imports|types_sizes|compiled_files|exports_file|name) took 513.544425ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 3.128213ms 
INFO [linters context/goanalysis] analyzers took 0s with no stages 
INFO [runner] Issues before processing: 16, after processing: 0 
INFO [runner] Processors filtering stat (out/in): autogenerated_exclude: 14/16, identifier_marker: 14/14, exclude: 14/14, filename_unadjuster: 16/16, skip_dirs: 16/16, cgo: 16/16, skip_files: 16/16, exclude-rules: 0/14, path_prettifier: 16/16 
INFO [runner] processing took 503.647µs with stages: identifier_marker: 198.697µs, path_prettifier: 150.633µs, skip_dirs: 63.793µs, autogenerated_exclude: 62.011µs, exclude-rules: 21.004µs, cgo: 2.489µs, filename_unadjuster: 1.479µs, max_same_issues: 841ns, nolint: 667ns, uniq_by_line: 346ns, source_code: 259ns, skip_files: 216ns, diff: 205ns, max_from_linter: 200ns, exclude: 177ns, severity-rules: 171ns, max_per_file_from_linter: 162ns, path_shortener: 118ns, sort_results: 98ns, path_prefixer: 81ns 
INFO [runner] linters took 178.694213ms with stages: goanalysis_metalinter: 178.087215ms 
INFO File cache stats: 0 entries of total size 0B 
INFO Memory: 8 samples, avg is 44.9MB, max is 99.4MB 
INFO Execution took 699.108716ms 

[etl@etl09 edgemesh]$ git branch
  ignore-rule-others
  main
* refresh-rules
  refresh-rules-in-ensure
  refresh-rules-interval
[etl@etl09 edgemesh]$ git push yz271544 refresh-rules -f
Enumerating objects: 68, done.
Counting objects: 100% (68/68), done.
Delta compression using up to 8 threads
Compressing objects: 100% (53/53), done.
Writing objects: 100% (56/56), 37.22 KiB | 9.30 MiB/s, done.
Total 56 (delta 13), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (13/13), completed with 10 local objects.
To github.com:yz271544/edgemesh-1.git
 + 0e19f14...db3374f refresh-rules -> refresh-rules (forced update)
```

Poorunga commented:
@yz271544 You can refer to: https://github.com/kubeedge/edgemesh/blob/main/hack/verify-vendor.sh

```
set -o errexit
set -o nounset
set -o pipefail

# The root of the build/dist directory
EDGEMESH_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"

function edgemesh::git::check_status() {
	# check if there's any uncommitted changes on go.mod, go.sum or vendor/
	echo $( git status --short 2>/dev/null | grep -E "go.mod|go.sum|vendor/" |wc -l)
}

${EDGEMESH_ROOT}/hack/update-vendor.sh
 
ret=$(edgemesh::git::check_status)
if [ ${ret} -eq 0 ]; then
	echo "SUCCESS: Vendor Verified."
else
	echo  "FAILED: Vendor Verify failed. Please run the command to check your directories: git status"
	exit 1
fi
```

**Check if there's any uncommitted changes on go.mod, go.sum or vendor**

khalid-jobs commented:
Do you try `./hack/update-vendor-licenses.sh` and `./hack/update-vendor.sh` ?

yz271544 commented:
[proxyby-edgemesh-update.log](https://github.com/kubeedge/edgemesh/files/7128604/proxyby-edgemesh-update.log)
I completely quit goland, and start the operation from clearing `$GOPATH/mod/*`. 
I will send out all the command line operations in Terminal, please help me to see if there is any problem?
:pray:

Poorunga commented:
Maybe we need a video conference

yz271544 commented:
Today, we conducted a video conference with community experts and determined that the short_if.go file format of the introduced third-party library go-funk was CRLF, which ultimately caused the verify-and-lint job abnormality of CI;
The processing log is now listed as follows:

1. modity `verify-vendor.sh`
```shell

set -o errexit
set -o nounset
set -o pipefail

# The root of the build/dist directory
EDGEMESH_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"

function edgemesh::git::check_status() {
	# check if there's any uncommitted changes on go.mod, go.sum or vendor/
	echo $( git status --short 2>/dev/null | grep -E "go.mod|go.sum|vendor/" |wc -l)
}

${EDGEMESH_ROOT}/hack/update-vendor.sh
 
ret=$(edgemesh::git::check_status)
if [ ${ret} -eq 0 ]; then
	echo "SUCCESS: Vendor Verified."
else
	git status  # Add a line of git status here, after the error is reported, the error log will be displayed
	echo  "FAILED: Vendor Verify failed. Please run the command to check your directories: git status"
	exit 1
fi
```

2. We perform dos2unix operation on the short_if.go file of go-funk locally, and re-execute `make verify` locally to reproduce the CI error reported on github.
```shell
[etl@etl09 proxy]$ gofmt -w iptables.go 
[etl@etl09 proxy]$ 
[etl@etl09 proxy]$ 
[etl@etl09 proxy]$ 
[etl@etl09 proxy]$ cd ../../
[etl@etl09 agent]$ cd ../
[etl@etl09 edgemesh-1]$ 
[etl@etl09 edgemesh-1]$ git status
On branch new-rules
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   agent/pkg/proxy/iptables.go
	modified:   go.mod
	modified:   go.sum
	new file:   vendor/github.com/thoas/go-funk/.gitignore
	new file:   vendor/github.com/thoas/go-funk/.travis.yml
	new file:   vendor/github.com/thoas/go-funk/CHANGELOG.md
	new file:   vendor/github.com/thoas/go-funk/LICENSE
	new file:   vendor/github.com/thoas/go-funk/Makefile
	new file:   vendor/github.com/thoas/go-funk/README.rst
	new file:   vendor/github.com/thoas/go-funk/assign.go
	new file:   vendor/github.com/thoas/go-funk/builder.go
	new file:   vendor/github.com/thoas/go-funk/chain_builder.go
	new file:   vendor/github.com/thoas/go-funk/compact.go
	new file:   vendor/github.com/thoas/go-funk/fill.go
	new file:   vendor/github.com/thoas/go-funk/go.mod
	new file:   vendor/github.com/thoas/go-funk/go.sum
	new file:   vendor/github.com/thoas/go-funk/helpers.go
	new file:   vendor/github.com/thoas/go-funk/intersection.go
	new file:   vendor/github.com/thoas/go-funk/join.go
	new file:   vendor/github.com/thoas/go-funk/join_primitives.go
	new file:   vendor/github.com/thoas/go-funk/lazy_builder.go
	new file:   vendor/github.com/thoas/go-funk/map.go
	new file:   vendor/github.com/thoas/go-funk/max.go
	new file:   vendor/github.com/thoas/go-funk/min.go
	new file:   vendor/github.com/thoas/go-funk/operation.go
	new file:   vendor/github.com/thoas/go-funk/options.go
	new file:   vendor/github.com/thoas/go-funk/permutation.go
	new file:   vendor/github.com/thoas/go-funk/predicate.go
	new file:   vendor/github.com/thoas/go-funk/presence.go
	new file:   vendor/github.com/thoas/go-funk/reduce.go
	new file:   vendor/github.com/thoas/go-funk/retrieve.go
	new file:   vendor/github.com/thoas/go-funk/scan.go
	new file:   vendor/github.com/thoas/go-funk/short_if.go
	new file:   vendor/github.com/thoas/go-funk/subset.go
	new file:   vendor/github.com/thoas/go-funk/subtraction.go
	new file:   vendor/github.com/thoas/go-funk/transform.go
	new file:   vendor/github.com/thoas/go-funk/typesafe.go
	new file:   vendor/github.com/thoas/go-funk/utils.go
	new file:   vendor/github.com/thoas/go-funk/without.go
	new file:   vendor/github.com/thoas/go-funk/zip.go
	modified:   vendor/modules.txt

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   agent/pkg/proxy/iptables.go

[etl@etl09 edgemesh-1]$ git diff
diff --git a/agent/pkg/proxy/iptables.go b/agent/pkg/proxy/iptables.go
index 14e1037..bfdee44 100644
--- a/agent/pkg/proxy/iptables.go
+++ b/agent/pkg/proxy/iptables.go
@@ -429,4 +429,3 @@ func (proxier *Proxier) CleanResidue() {
                }
        }
 }
-
[etl@etl09 edgemesh-1]$ git add .
[etl@etl09 edgemesh-1]$ 
[etl@etl09 edgemesh-1]$ 
[etl@etl09 edgemesh-1]$ make lint
hack/make-rules/lint.sh
checking golangci-lint
found golangci-lint, version: golangci-lint has version 1.42.1 built from 54f4301d on 2021-09-06T17:05:10Z
start lint ...
check any whitenoise ...
check any issue by golangci-lint ...
INFO [config_reader] Config search paths: [./ /tmp/edgemesh-1 /tmp / /home/etl] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 8 linters: [errcheck gci goconst gofmt gosimple govet misspell whitespace] 
INFO [loader] Go packages loading at mode 575 (deps|exports_file|imports|name|types_sizes|compiled_files|files) took 652.704875ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 2.74638ms 
INFO [linters context/goanalysis] analyzers took 395.79984ms with top 10 stages: buildir: 367.962839ms, printf: 8.055789ms, gofmt: 3.568421ms, S1038: 2.563008ms, misspell: 1.874186ms, ctrlflow: 1.317539ms, S1039: 1.10382ms, inspect: 819.116µs, whitespace: 814.84µs, S1025: 634.782µs 
INFO [runner] Issues before processing: 16, after processing: 0 
INFO [runner] Processors filtering stat (out/in): autogenerated_exclude: 14/16, exclude-rules: 0/14, cgo: 16/16, filename_unadjuster: 16/16, skip_dirs: 16/16, identifier_marker: 14/14, exclude: 14/14, path_prettifier: 16/16, skip_files: 16/16 
INFO [runner] processing took 410.709µs with stages: identifier_marker: 199.353µs, path_prettifier: 86.198µs, autogenerated_exclude: 54.034µs, skip_dirs: 42.704µs, exclude-rules: 21.043µs, cgo: 2.168µs, filename_unadjuster: 1.673µs, nolint: 807ns, max_same_issues: 550ns, uniq_by_line: 398ns, max_from_linter: 298ns, source_code: 254ns, exclude: 195ns, diff: 181ns, skip_files: 178ns, severity-rules: 171ns, max_per_file_from_linter: 156ns, path_shortener: 152ns, sort_results: 116ns, path_prefixer: 80ns 
INFO [runner] linters took 812.42282ms with stages: goanalysis_metalinter: 811.944161ms 
INFO File cache stats: 6 entries of total size 26.6KiB 
INFO Memory: 16 samples, avg is 149.9MB, max is 317.5MB 
INFO Execution took 1.471685929s                  
[etl@etl09 edgemesh-1]$ 
[etl@etl09 edgemesh-1]$ 
[etl@etl09 edgemesh-1]$ make verify
hack/verify-golang.sh
go detail version: go version go1.14.15 linux/amd64
go version: 1.14.15
hack/verify-vendor.sh
running 'go mod tidy'
running 'go mod vendor'
FAILED: Vendor Verify failed. Please run the command to check your directories: git status
make: *** [Makefile:77: verify-vendor] Error 1
[etl@etl09 edgemesh-1]$ git status
On branch new-rules
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   agent/pkg/proxy/iptables.go
	modified:   go.mod
	modified:   go.sum
	new file:   vendor/github.com/thoas/go-funk/.gitignore
	new file:   vendor/github.com/thoas/go-funk/.travis.yml
	new file:   vendor/github.com/thoas/go-funk/CHANGELOG.md
	new file:   vendor/github.com/thoas/go-funk/LICENSE
	new file:   vendor/github.com/thoas/go-funk/Makefile
	new file:   vendor/github.com/thoas/go-funk/README.rst
	new file:   vendor/github.com/thoas/go-funk/assign.go
	new file:   vendor/github.com/thoas/go-funk/builder.go
	new file:   vendor/github.com/thoas/go-funk/chain_builder.go
	new file:   vendor/github.com/thoas/go-funk/compact.go
	new file:   vendor/github.com/thoas/go-funk/fill.go
	new file:   vendor/github.com/thoas/go-funk/go.mod
	new file:   vendor/github.com/thoas/go-funk/go.sum
	new file:   vendor/github.com/thoas/go-funk/helpers.go
	new file:   vendor/github.com/thoas/go-funk/intersection.go
	new file:   vendor/github.com/thoas/go-funk/join.go
	new file:   vendor/github.com/thoas/go-funk/join_primitives.go
	new file:   vendor/github.com/thoas/go-funk/lazy_builder.go
	new file:   vendor/github.com/thoas/go-funk/map.go
	new file:   vendor/github.com/thoas/go-funk/max.go
	new file:   vendor/github.com/thoas/go-funk/min.go
	new file:   vendor/github.com/thoas/go-funk/operation.go
	new file:   vendor/github.com/thoas/go-funk/options.go
	new file:   vendor/github.com/thoas/go-funk/permutation.go
	new file:   vendor/github.com/thoas/go-funk/predicate.go
	new file:   vendor/github.com/thoas/go-funk/presence.go
	new file:   vendor/github.com/thoas/go-funk/reduce.go
	new file:   vendor/github.com/thoas/go-funk/retrieve.go
	new file:   vendor/github.com/thoas/go-funk/scan.go
	new file:   vendor/github.com/thoas/go-funk/short_if.go
	new file:   vendor/github.com/thoas/go-funk/subset.go
	new file:   vendor/github.com/thoas/go-funk/subtraction.go
	new file:   vendor/github.com/thoas/go-funk/transform.go
	new file:   vendor/github.com/thoas/go-funk/typesafe.go
	new file:   vendor/github.com/thoas/go-funk/utils.go
	new file:   vendor/github.com/thoas/go-funk/without.go
	new file:   vendor/github.com/thoas/go-funk/zip.go
	modified:   vendor/modules.txt

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   vendor/github.com/thoas/go-funk/short_if.go
```

solution:
1. Submit pr to go-funk and hope they can solve this bug as soon as possible;
2. Or we can use alternatives to replace go-funk;

Poorunga commented:
/lgtm

yz271544 commented:
modify proxyby!=edgemesh to noproxy=edgemesh

yz271544 commented:
modify the ignore log tip:
```go
klog.V(4).Infof("ignored containing noproxy=edgemesh label service: %s", otherIgnoreService.Name)
```

Poorunga commented:
/lgtm
